Sources/CoreAPI/WordPressComRestApi.swift

```diff
@@ -152,12 +152,6 @@ open class WordPressComRestApi: NSObject {
         super.init()
     }
 
-    deinit {
-        for session in [urlSession, uploadURLSession] {
-            session.finishTasksAndInvalidate()
-        }
-    }
```

Comment:
One side effect is ongoing tasks won't be invalidated at all. I wonder if we should make `WordPressComRestApi` accept a URLSession instance, which the app can create upon use signing in and invalidate upon logout.

Reply:
Is there a need to invalidate it? The requests are signed on a per-request basis now.

Something does need to change with the `WordPressComRestApi/invalidateAndCancelTasks` usage. The lifetime of `_wordPressComRestApi` is tied to a `WPAccount`, which is also problematic: there is one instance per thread, and the instances could get reset.

Reply:
> Is there a need to invalidate it?

Good question. I don't know for sure though. Using a shared URLSession singleton between user sessions is a change introduced in this PR, which we'll need to test to make sure it doesn't break things.

IMO, the best practice is URLSession instance should be per user session. Upon user logs in, they get a new URLSession instance and uses it until logs out. Once logs out, the URLSession instance should be destroyed. That's similar to how Safari uses one URLSession for each tab.

> The lifetime of _wordPressComRestApi is tied to a WPAccount, which is also problematic: there is one instance per thread, and the instances could get reset.

Yes, definitely. That's another big piece of tech debt.

Reply:
> IMO, the best practice is URLSession instance should be per user session. Upon user logs in, they get a new URLSession instance and uses it until logs out. Once logs out, the URLSession instance should be destroyed. That's similar to how Safari uses one URLSession for each tab.

I completely agree. The app should have a repository that could control the lifetime of the objects to match the lifetime of the user session. It's probably a better fix for this issue as well (assuming it actually fixes it because I don't have any hard evidence yet).